bindings/java/src/org/sleuthkit/datamodel/SleuthkitCase.java

```diff
@@ -475,45 +483,70 @@ private void updateDatabaseSchema(String dbPath) throws Exception {
+				dbSchemaMajorVersion = resultSet.getInt("schema_ver"); //NON-NLS
+				if (hasMinorVersion) {
+					//if there is a minor version column, use it, else default to zero.
+					dbSchemaMinorVersion = resultSet.getInt("schema_minor_ver"); //NON-NLS
+				}
+			} else {
+				throw new TskCoreException();
 			}
+			CaseDBSchemaVersionNumber dbSchemaVersion = new CaseDBSchemaVersionNumber(dbSchemaMajorVersion, dbSchemaMinorVersion);
+
 			resultSet.close();
 			resultSet = null;
 			statement.close();
 			statement = null;
 
-			// Do the schema update(s), if needed.
-			if (SCHEMA_VERSION_NUMBER != schemaVersionNumber) {
+			//check schema compatibility
+			if (false == CURRENT_DB_SCHEMA_VERSION.isCompatible(dbSchemaVersion)) {
+				//we cannot open a db with a major schema version higher than the current one.
+				throw new TskUnsupportedSchemaVersionException(
+						"Unsupported DB schema version " + dbSchemaVersion + ", the highest supported schema version is " + CURRENT_DB_SCHEMA_VERSION.getMajor() + ".X");
+			} else if (dbSchemaVersion.compareTo(CURRENT_DB_SCHEMA_VERSION) < 0) {
+				//The schema version is compatible,possibly after upgrades.
+
 				if (null != dbPath) {
 					// Make a backup copy of the database. Client code can get the path of the backup
 					// using the getBackupDatabasePath() method.
-					String backupFilePath = dbPath + ".schemaVer" + schemaVersionNumber + ".backup"; //NON-NLS
+					String backupFilePath = dbPath + ".schemaVer" + dbSchemaVersion.toString() + ".backup"; //NON-NLS
 					copyCaseDB(backupFilePath);
 					dbBackupPath = backupFilePath;
 				}
 
 				// ***CALL SCHEMA UPDATE METHODS HERE***
-				// Each method should examine the schema number passed to it and either:
-				//    a. do nothing and return the schema version number unchanged, or
-				//    b. upgrade the database and then increment and return the schema version number.
-				schemaVersionNumber = updateFromSchema2toSchema3(schemaVersionNumber, connection);
-				schemaVersionNumber = updateFromSchema3toSchema4(schemaVersionNumber, connection);
-				schemaVersionNumber = updateFromSchema4toSchema5(schemaVersionNumber, connection);
-				schemaVersionNumber = updateFromSchema5toSchema6(schemaVersionNumber, connection);
-				schemaVersionNumber = updateFromSchema6toSchema7(schemaVersionNumber, connection);
+				// Each method should examine the schema version passed to it and either:
+				//    a. do nothing and return the schema version unchanged, or
+				//    b. upgrade the database and return the schema version that the db was upgraded to.
+				dbSchemaVersion = updateFromSchema2toSchema3(dbSchemaVersion, connection);
+				dbSchemaVersion = updateFromSchema3toSchema4(dbSchemaVersion, connection);
+				dbSchemaVersion = updateFromSchema4toSchema5(dbSchemaVersion, connection);
+				dbSchemaVersion = updateFromSchema5toSchema6(dbSchemaVersion, connection);
+				dbSchemaVersion = updateFromSchema6toSchema7dot1(dbSchemaVersion, connection);
```

Comment:
Why 7.1 instead of 7.0?

Reply:
Gah, I can't remember the details, but there was a reason.  It had something to do with wanting to be able to upgrade existing "dev" cases that already had a schema 7 in them.  Maybe we could stay at 7.0 if tat matters.

Reply:
As you have already seen, I had Ann restore the original 6 to 7 upgrade, and leave your new code in your 7 to 7.1 upgrade, which makes it easy to both upgrade our plentiful supply of "dev" cases and "real world" cases when we release.  It thinking this was what you intended, and got side-tracked. Stuff happens., it's all good now.